moirae/estimators/online/dual.py

```diff
@@ -61,20 +66,41 @@ def __init__(self, transient_filter: BaseFilter, asoh_filter: BaseFilter) -> Non
         self.cell_wrapper = cell_wrapper
         self.asoh_wrapper = asoh_wrapper
 
-    @property
-    def state(self) -> MultivariateRandomDistribution:
+    def _get_converted_states(self) -> Tuple[MultivariateRandomDistribution, MultivariateRandomDistribution]:
+        """
+        Helper function to convert values in transient and A-SOH from filter representations
+
+        Returns:
+            transient_converted: representation of the transient
+                :class:`~moirae.estimators.online.filters.distributions.MultivariateRandomDistribution` in
+                model-coordinates
+            asoh_converted: representation of the A-SOH
+                :class:`~moirae.estimators.online.filters.distributions.MultivariateRandomDistribution` in
+                model-coordinates
+        """
         # We need to get the hidden states on both filters
         transient_hidden = self.trans_filter.hidden.model_copy(deep=True)
         asoh_hidden = self.asoh_filter.hidden.model_copy(deep=True)
-        return transient_hidden.combine_with(asoh_hidden)
+
+        # We need to transform them accordingly
+        transient_converted = transient_hidden.convert(conversion_operator=self.trans_filter.model.hidden_conversion)
```

Comment:
Do we need the copy anymore if the the `convert` method produces a new instance of the distribution?

Reply:
You are right, no need to crowd memory. What do you think of this new version? (btw, I checked to make sure they run fine in the slow tests locally)

Reply:
I was more pointing out the `model_copy` statements in the previous lines. You shouldn't need to copy and then convert, just convert.